doc/Module 9 part 3 - classifying timestamps.md

```diff
@@ -0,0 +1,315 @@
+always a not null value. But `facilityTypeCode` can be null in some cases (e.g., `EOSP`).
+
+When there is a `portCallServiceTypeCode`, then `facilityTypeCode` is _often_ `BRTH` as
+_most_ services only happen at berth. However, a few services such as Bunkering or Sludge,
+which _can_ happen at anchorage (`facilityTypeCode` with value of `ANCH`).  Additionally,
+some timestamps (based on the `portCallPhaseTypeCode`) can happen at Pilot Boarding
+Place (`facilityTypeCode` with value of `PBPL`).
+
+As noted in the previous parts, the `facilityTypeCode` also affects the `Event Location Info`
+fields.  When `facilityTypeCode` is:
+
+ * `BRTH` then the location must include a terminal (see the attributes starting with `facility`).
+   If a `locationName` is provided, it defines the name of the berth.
+ * `PBPL` then the location must *not* include a terminal.  The `locationName` then refers to
+   the pilot boarding place.
+ * `ANCH` then the location must *not* include a terminal.  The `locationName` then refers to
+   the anchorage location.
+
+With the addition of the `facilityTypeCode`, you can now represent the following extra
+timestamps:
+
+ * `ETA-PBP`
+ * `RTA-PBP`
+ * `PTA-PBP`
+ * `ATA-PBP`
+ * `ETA-Anchorage`
+ * `RTA-Anchorage`
+ * `PTA-Anchorage`
+ * `ATA-Anchorage`
+
+
+To conclude this subsection with a concrete example, here is a `ETA-PBP`:
+
+```json
+{
+  "eventClassifierCode": "EST",
+  "operationsEventTypeCode": "ARRI",
+  "facilityTypeCode": "PBPL",
+  "portCallPhaseTypeCode": null,
+  "portCallServiceTypeCode": null
+}
+```
+
+Note that only the `facilityTypeCode` changed compared to the `ETA-Berth`.
+
+# The `portCallServiceTypeCode`  attribute
+
+The `portCallServiceTypeCode` defines which service the timestamp is about. When the timestamp
+follows the nice pattern, then the `portCallServiceTypeCode` also contributes the name of the
+timestamp.
+
+It is worth noting that services use `STRT` (Start) / `CMPL` (Complete) rather than `ARRI`
+(Arrival) / `DEPA` (Departure) in the `operationsEventTypeCode`.  This also means that all the
+service related timestamps that follow the "nice pattern" use `S` or `C` in their third letter
+(such as `ETS-Cargo Ops`).
+
+The `portCallServiceTypeCode` has a long list of values, this guide will not cover all them.
+Instead, please review the Swagger specification or the IFS for the full list.  For now, here
+is an incomplete list for the sake of the example:
+
+ * `CRGO` - Cargo Ops (`ETS-Cargo Ops`)
+ * `BUNK` - Bunkering (`ETS-Bunkering`)
+ * `SLUG` - Sludge (`ETS-Sludge`)
+
+The above here do not require `portCallPhaseTypeCode` and with these you can now start to specify
+the following timestamps:
+
+ * `ETS-Cargo Ops`
+ * `RTS-Cargo Ops`
+ * `PTS-Cargo Ops`
+ * `ATS-Cargo Ops`
+ * `ETC-Cargo Ops`
+ * `RTC-Cargo Ops`
+ * `PTC-Cargo Ops`
+ * `ATC-Cargo Ops`
+
+Rinse and repeat for the other service types listed above.  As an example, here is a `RTC-Cargo Ops`:
+
+```json
```

Comment:
what distinguishes the below example from an ETS-Cargo Ops?

Reply:
The JSON was in fact an `ETC-Cargo Ops` rather than an `RTC-Cargo Ops` (with invalid `facilityTypeCode`).  I have revised it to be an `ETS-Cargo Ops`